tools/codegen/src/ast/verified/mod.rs

```diff
@@ -115,6 +115,30 @@ pub struct ItemDecl {
     typ: Rc<TopDecl>,
 }
 
+#[derive(Debug, Property)]
+#[property(get(public))]
+pub struct UnionItemDecl {
+    typ: Rc<TopDecl>,
+    id: usize,
+}
+
+impl From<(ItemDecl, usize)> for UnionItemDecl {
+    fn from((item, id): (ItemDecl, usize)) -> Self {
+        Self { typ: item.typ, id }
+    }
+}
+
+impl UnionItemDecl {
+    pub fn split(&self) -> (ItemDecl, usize) {
+        (
+            ItemDecl {
+                typ: Rc::clone(&self.typ),
+            },
+            self.id,
+        )
+    }
+}
+
```

Comment:
I think these two fn can be removed since they are not used in latest codebase

Reply:
Thanks for your suggestions. I have pushed a commit to apply these.